packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -213,31 +221,49 @@ - (void)enableContextMenu
 {
   _longPressGestureRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self
                                                                               action:@selector(handleLongPress:)];
+  if (@available(iOS 16.0, *)) {
+      _editMenuInteraction = [[UIEditMenuInteraction alloc] initWithDelegate:self];
+      [self addInteraction:_editMenuInteraction];
+  }
+    
   [self addGestureRecognizer:_longPressGestureRecognizer];
 }
 
 - (void)disableContextMenu
 {
   [self removeGestureRecognizer:_longPressGestureRecognizer];
+    
+  if (@available(iOS 16.0, *)) {
+    [self removeInteraction:_editMenuInteraction];
+    _editMenuInteraction = nil;
+  }
   _longPressGestureRecognizer = nil;
 }
 
 - (void)handleLongPress:(UILongPressGestureRecognizer *)gesture
 {
-  // TODO: Adopt showMenuFromRect (necessary for UIKitForMac)
 #if !TARGET_OS_UIKITFORMAC
-  UIMenuController *menuController = [UIMenuController sharedMenuController];
+  if (@available(iOS 16.0, *)) {
+      CGPoint location = [gesture locationInView:self];
+      UIEditMenuConfiguration *config = [UIEditMenuConfiguration configurationWithIdentifier:nil sourcePoint:location];
+      if (_editMenuInteraction) {
+          [_editMenuInteraction presentEditMenuWithConfiguration:config];
+      }
+  } else {
```

Comment:
nit: can we do something like
```objc
if (@available(iOS 16.0, *)) {
  // your code
  return
}
// old code
```
a part from me, preferring early returns when reading code, it will also make the diff more readable as the body of the now non-existing `else` will not be indented and github will not highlight it as changed code.

Reply:
Thanks for the review! I've applied your suggestion. I saw that linter step failed - I'm curious what do you use to format Objective-C code in Xcode?

Reply:
It's... a complicated question.
We use the same tool we have in OSS, but:
- the version we use internally have been modified by some other Meta engineer
- the set of rules we use internally are different and spread in different places, so they are very hard to retrieve.

We already tried several times to align the internal and external signals, but it is a harder problem than it look... :(

Reply:
Ah okay I get it, thanks for the explanation